design-documents/move-specific-layout-edit.md

```diff
@@ -0,0 +1,47 @@
+### Move category, product and page specific layout updates
+Custom layout updates can be provided for specific categories, products and CMS pages via respective forms in the admin
+panel - this is how Merchants can customize entity-specific pages to include additional design and functionality.
+ 
+![alt text](img/custom_alyout_updates.PNG "Layout updates form")
+### The problem
+Layout updates are defined using XML and certain instructions often using PHP classes. Only developers can
+posses knowledge required to create a layout update yet the only way to provide these updates
+is via user forms more commonly used by content managers. That's the __1st problem__ - content managers don't
+need that functionality and it is better to hide it from them to avoid them from executing PHP code on the server.
+ 
+__2nd problem__ is that we do not allow developers to work with entity-specific layout updates in a developer way.
+Meaning that there is no way to use version control while editing these layout updates.
+
+### Solution
+Read custom layout updates from physical files on the server. Convert existing textarea _custom_layout_update_ field
+in category/product/cms page form into a _select_ field where existing custom layout update files could be selected.
+A developer would create a layout file following this template:
+_app/design/custom\_layout/\<category/product/page\>/layout\_update\_\<entity ID\>\_\<readable layout update name\>.xml_.
```

Comment:
It's confusing that in case of a single layout update, a developer can use old approach with `catalog_product_view_id_128.xml`, while as soon as it's necessary to support multiple templates, a developer should put the layout updates in an absolutely different folder (`app/design/custom_layout/page/layout_update_128_store1update.xml`). Btw, the rule for forming the name is not very clear.  There is no correlation with default layout update (`cms_page_view.xml`). IMO, it will be hard to find all applied layouts intuitively.
Also, it introduces new concept - `custom_layout` is not a theme, though it looks like a theme based on the location. We already have `base`, which is confusing in the same way.

I propose to consider an option of reusing existing functionality by extending it so it supports multiple layouts that can be selected by an admin.
For example, in addition to `catalog_product_view_id_128.xml`, also support:
1. `catalog_product_view_id_128/layout1.xml`, `catalog_product_view_id_128/layout2.xml`
1. or `catalog_product_view_id_128_layout1.xml`, `catalog_product_view_id_128_layout2.xml`

Location will be the same as supported now: a custom module(s) or a custom theme(s), whatever works best for specific case.

Reply:
It was brought to my attention that this functionality may be used by merchants without a dedicated development team and was trying to make it easier for them to create these layout updates files. It may be a bit harder for such merchants to register a new module/theme. But I don't mind using existing target directories for these updates, it would be better not to introduce too many new conventions. @piotrekkaminski What do you think?
Same goes to naming, since these custom updates are meant to replace the updates from the admin panel which only allows to alter layouts for 'view' pages I didn't think it was necessary to follow these conventions which allow to also redefine admin 'Edit' pages. But again I understand the value of preserving existing conventions as much as possible

Reply:
i'm ok with this cahnge

Reply:
Further I think it would be better to have old way single-entity updates and these new updates separately - when a developer create a catalog_product_view_id_128.xml style file they won't expect it to be controlled by content managers via admin panel